### PR TITLE
[13.0][IMP] stock_request: Stock Request filters + test extensibility

### PR DESCRIPTION
**Pull request by HviorForgeFlow**
- [x]  Add search filters to Stock Request
- [x]  Use SR Manager on action_confirm from testing for better extensibility

CC @ForgeFlow @etobella @jarroyomorales

### CODE REVIEW

**Review by ps-tubtim — APPROVED** [review]
Code LGTM.

**Review by AaronHForgeFlow — APPROVED** [review]
:+1:  LGTM

File: stock_request/tests/test_stock_request.py [hunk 1]
```diff
@@ -734,7 +734,7 @@ def test_cancel_request(self):
         order = self.request_order.with_user(self.stock_request_user).create(vals)
 
         self.product.route_ids = [(6, 0, self.route.ids)]
-        order.with_user(self.stock_request_user).action_confirm()
+        order.with_user(self.stock_request_manager).action_confirm()
```
Comment (LoisRForgeFlow) [inline stock_request/tests/test_stock_request.py:737]: why is this needed? test were working before this improvement, what have changed?

Comment (HviorForgeFlow) [reply stock_request/tests/test_stock_request.py:737]: Not in this module, but stock_request_user don't belong to inventory users, so for example when mrp is installed and tests run I've got several errors regarding read access that cannot be avoidable during installation adding access rules on extensible module.

Comment (LoisRForgeFlow) [reply stock_request/tests/test_stock_request.py:737]: fair enough

**Comment by LoisRForgeFlow** [comment]
Good to go!

/ocabot merge minor
